File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DepartmentsRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface DepartmentsRow {
    DepartmentId?: number;
    Name?: string;
}

export abstract class DepartmentsRow {
    static readonly idProperty = 'DepartmentId';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Tanimlamalar.Departments';
    static readonly lookupKey = 'Tanimlamalar.Departments';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<DepartmentsRow>('Tanimlamalar.Departments') }
    static async getLookupAsync() { return getLookupAsync<DepartmentsRow>('Tanimlamalar.Departments') }

    static readonly deletePermission = 'Tanimlamalar.Departments.General';
    static readonly insertPermission = 'Tanimlamalar.Departments.General';
    static readonly readPermission = 'Tanimlamalar.Departments.General';
    static readonly updatePermission = 'Tanimlamalar.Departments.General';

    static readonly Fields = fieldsProxy<DepartmentsRow>();
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Departments/DepartmentsRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar;

[ConnectionKey("Default"), Module("Tanimlamalar"), TableName("Departments")]
[DisplayName("Departments"), InstanceName("Departments")]
[ReadPermission("Tanimlamalar.Departments.General")]
[ModifyPermission("Tanimlamalar.Departments.General")]
[ServiceLookupPermission("Tanimlamalar.Departments.General")]
[LookupScript]
public sealed class DepartmentsRow : Row<DepartmentsRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Department Id"), Identity, IdProperty]
    public int? DepartmentId { get => fields.DepartmentId[this]; set => fields.DepartmentId[this] = value; }

    [DisplayName("Name"), Size(200), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field DepartmentId;
        public StringField Name;

    }
}